osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmap.cs

```diff
@@ -212,10 +264,40 @@ protected override void ApplyState()
                 starDifficultyBindable = difficultyCache.GetBindableDifficulty(beatmapInfo, (starDifficultyCancellationSource = new CancellationTokenSource()).Token);
                 starDifficultyBindable.BindValueChanged(d =>
                 {
-                    starCounter.Current = (float)(d.NewValue?.Stars ?? 0);
-                    if (d.NewValue != null)
-                        difficultyIcon.Current.Value = d.NewValue.Value;
+                    starRatingDisplay.Current.Value = d.NewValue ?? default;
                 }, true);
+
+                starRatingDisplay.Current.BindValueChanged(s =>
+                {
+                    starCounter.Current = (float)s.NewValue.Stars;
+
+                    // Every other element in song select that uses this cut off uses yellow for the upper range but the designs use white here for whatever reason.
+                    iconContainer.Colour = s.NewValue.Stars > 6.5f ? Colour4.White : colourProvider.Background5;
+
+                    var starRatingColour = colours.ForStarDifficulty(s.NewValue.Stars);
+
+                    starCounter.Colour = colourBox.Colour = starRatingColour;
+                    starRatingGradient.Colour = ColourInfo.GradientHorizontal(starRatingColour.Opacity(0.25f), starRatingColour.Opacity(0));
+                    starRatingGradient.Show();
+
+                    if (Item!.State.Value == CarouselItemState.NotSelected) return;
+
+                    // We want to update the EdgeEffect here instead of in selected() to make sure the colours are correct
+                    Header.EffectContainer.EdgeEffect = new EdgeEffectParameters
+                    {
+                        Type = EdgeEffectType.Shadow,
+                        Colour = starCounter.Colour.MultiplyAlpha(0.5f),
+                        Radius = 50
```

Comment:
This might be a performance issue, and i'm not sure how good this looks visually either.